coreax/approximation.py

```diff
@@ -338,11 +338,75 @@ def approximate(
         :return: Approximation of the kernel matrix inverse
         """
 
+    def _map_approximate(self) -> Callable[[Array, float, Array], Array]:
+        """Define helper function to map approximate over horizontal array stack."""
+        return jax.vmap(self.approximate, in_axes=(0, None, None))
+
+    def approximate_stack(
+        self, kernel_gramians: Array, regularisation_parameter: float, identity: Array
+    ) -> Array:
+        r"""
+        Approximate the regularised inverses of a horizontal stack of kernel matrices.
+
+        :param kernel_gramian: Horizontal Stack of :math:`n \times n` kernel gram
+            matrices
+        :param regularisation_parameter: Regularisation parameter for stable inversion
+            of array, negative values will be converted to positive
+        :param identity: Block identity matrix
+        :return: Approximation of the kernel matrix inverses
+        """
+        return self._map_approximate()(
+            kernel_gramians, regularisation_parameter, identity
+        )
+
+
+class LeastSquareApproximator(RegularisedInverseApproximator):
+    """
+    Approximate inverse of regularised gramian by solving a least-squares problem.
+
+    Note that this approximator does not give time savings and instead acts as a
+    default option useful for comparing other approximators to.
+
+    :param random_key: Key for random number generation
+    :param rcond: Cut-off ratio for small singular values of 'array'. For the purposes
+        of rank determination, singular values are treated as zero if they are smaller
+        than rcond times the largest singular value of 'array'. The default value of
+        None will use the machine precision multiplied by the largest dimension of the
+        array. An alternate value of -1 will use machine precision.
+    """
+
+    def __init__(self, random_key: KeyArrayLike, rcond: Optional[float] = None):
+        """Initialise LeastSquareApproximator class and validate input."""
+        # Initialise parent
+        super().__init__(random_key=random_key)
+        self.rcond = rcond
+
+        # Validate input
+        if self.rcond is not None:
+            if self.rcond < 0 and self.rcond != -1:
+                raise ValueError("'rcond' must be non-negative, except for value of -1")
+
+    @override
+    def approximate(
+        self,
+        kernel_gramian: Array,
+        regularisation_parameter: float,
+        identity: Array,
+    ) -> Array:
+        if kernel_gramian.shape != identity.shape:
```

Comment:
The JAX error raised here is essentiallly the same, "Leading dimensions of input arrays must match". If you want to name these arrays, then you should utilise a try-except here, instead of an explicit if statement. This try-except pattern is consitent with the rest of the codebase and is generally preferd for two reasons. 1. It allows us to be more permisive towards user input, only raising when the underlying problem is triggered. 2. It yields a more informative stack-trace; the current approach will tell the user the leading dimensions need to match, but won't make clear why they need to match; a try-except approach would raise from the underlying error raised in `lstsq`, making it more obvious why this dimension mismatch is a problem.